pkgs/development/cuda-modules/_cuda/db/schema.nix

```diff
@@ -0,0 +1,608 @@
+{ lib, config, ... }:
+let
+  inherit (lib) mkOption types literalExpression;
+  inherit (lib.types)
+    attrsOf
+    bool
+    enum
+    listOf
+    nullOr
+    str
+    submodule
+    mkOptionType
+    ;
+
+  inherit (import ./columnar.nix)
+    Unit
+    SetOfStr
+    mkColumnOption
+    assertSubset
+    assertCompleteTable
+    ;
+
+  bootstrapData = import ./bootstrap { inherit lib; } // {
+    _file = ./bootstrap;
+  };
+  mergeByLength =
+    getLength: loc: defs:
+    if defs == [ ] then
+      abort "This case should never happen."
+    else if lib.length defs == 1 then
+      (lib.head defs).value
+    else
+      (lib.foldl' (
+        a: b:
+        let
+          aLen = getLength a.value;
+          bLen = getLength b.value;
+        in
+        if a.value == b.value then
+          a
+        else if aLen > bLen then
+          a
+        else if bLen > aLen then
+          b
+        else
+          throw "The option `${lib.showOption loc}' has conflicting definition values of same length:${
+            lib.showDefs [
+              a
+              b
+            ]
+          }"
+      ) (lib.head defs) (lib.tail defs)).value;
+  StrByLength = mkOptionType {
+    name = "StrByLength";
+    description = "string (merged by choosing longest)";
+    descriptionClass = "noun";
+    check = lib.isString;
+    merge = mergeByLength lib.stringLength;
+  };
```

Comment:
Is there a way to avoid the need for this type? I worry that equating priority to length is a bad design choice.

Reply:
I used to `mkOverride` with with different priorities and then dropped that... forget why. But here the reason is just to make merging of NVIDIA jsons with conflicting values more deterministic.

Reply:
Ah, okay.

Do you happen to remember any examples of conflicting values from their JSON? I'm interested, and I think Kevin would be as well.